### PR TITLE
Uncrosser improvements

### PR DESCRIPTION
**Pull request by NourAlharithi**

### CODE REVIEW

File: src/bots/uncrossArbBot.ts [hunk 1]
```diff
@@ -324,4 +572,70 @@ export class UncrossArbBot implements Bot {
 			}
 		}
 	}
+
+	private async settlePnls() {
```
Comment (crispheaney) [inline src/bots/uncrossArbBot.ts:576]: we should recursively try to call settle pnls in smaller chunks if they fail i think ... we should probably generalize that pattern to so we don't have to keep implementing it, i can do that

File: src/bots/uncrossArbBot.ts [hunk 2]
```diff
@@ -212,45 +437,47 @@ export class UncrossArbBot implements Bot {
...
-
-					console.log('best ask', bestDriftAsk.userAccount!.toBase58());
-					console.log('best bid', bestDriftBid.userAccount!.toBase58());
-
 					const midPrice = (bestBidPrice + bestAskPrice) / 2;
 					if (
 						(bestBidPrice - bestAskPrice) / midPrice >
 						2 * driftUser.getMarketFees(MarketType.PERP, perpIdx).takerFee
 					) {
+						let bidMakerInfo: MakerInfo;
```
Comment (crispheaney) [inline src/bots/uncrossArbBot.ts:445]: think we should log the market, the users, there size and price in here so it's easier to debug

File: src/utils.ts [hunk 3]
```diff
@@ -199,7 +208,14 @@ export function getBestLimitAskExcludePubKey(
 		oraclePriceData
 	);
 	for (const ask of asks) {
-		if (ask.userAccount?.equals(excludedUserAccount)) {
+		if (
+			ask.userAccount?.toBase58() === excludedPubKey ||
+			excludedUserAccountsAndOrder?.some(
```
Comment (crispheaney) [inline src/utils.ts:213]: might be nice to put this on a separate if block so it's more readable

File: src/bots/uncrossArbBot.ts [hunk 4]
```diff
@@ -159,10 +204,168 @@ export class UncrossArbBot implements Bot {
...
-	 * Typical bot loop that runs periodically and pats the watchdog timer on completion.
-	 *
-	 */
+	protected getOrderSignatureFromMakerInfo(makerInfo: MakerInfo): string {
+		return (
+			makerInfo.maker.toString() + '_' + makerInfo.order!.orderId.toString()
+		);
+	}
+
+	protected handleTransactionLogs(
```
Comment (crispheaney) [inline src/bots/uncrossArbBot.ts:213]: this a good candidate for unit test tbh

File: src/bots/uncrossArbBot.ts (hunk 2 above)
Comment (NourAlharithi) [reply src/bots/uncrossArbBot.ts:445]: done

File: src/utils.ts (hunk 3 above)
Comment (NourAlharithi) [reply src/utils.ts:213]: done

**Pull request merged**